src/main/java/org/broadinstitute/hellbender/engine/LocusWalker.java

```diff
@@ -121,8 +136,12 @@ public void traverse() {
         CountingReadFilter countedFilter = disableAllReadFilters ?
                 new CountingReadFilter("Allow all", ReadFilterLibrary.ALLOW_ALL_READS ) :
                 makeReadFilter();
+        // get the transformer
+        ReadTransformer transformer = makeReadTransformer();
+        // get the transformed iterator
+        final Iterator<GATKRead> readIterator = StreamSupport.stream(reads.spliterator(), false).map(transformer).iterator();
         // get the LIBS
-        LocusIteratorByState libs = new LocusIteratorByState(new ReadFilteringIterator(reads.iterator(), countedFilter), getDownsamplingMethod(), includeDeletions(), includeNs(), keepUniqueReadListInLibs(), samples, header);
+        LocusIteratorByState libs = new LocusIteratorByState(new ReadFilteringIterator(readIterator, countedFilter), getDownsamplingMethod(), includeDeletions(), includeNs(), keepUniqueReadListInLibs(), samples, header);
```

Comment:
Can you document in the class-level docs for `LocusWalker` that read transformers are applied *before* read filters?